api/apiroutes.go

```diff
@@ -171,29 +171,66 @@ func NewContext(cfg *AppContextConfig) *appContext {
 	}
 }
 
+func (c *appContext) updateNodeConnections() error {
+	nodeConnections, err := c.nodeClient.GetConnectionCount()
+	if err != nil {
+		// Assume there arr no connections if RPC had an error.
+		nodeConnections = 0
+		err = fmt.Errorf("failed to get connection count: %v", err)
+	}
+
+	// Before updating connections, get the previous connection count.
+	prevConnections := c.Status.NodeConnections()
+
+	c.Status.SetConnections(nodeConnections)
+	if nodeConnections == 0 {
+		return err
```

Comment:
I actually confused myself while working on this, but it can be non-nil because the `if err != nil {` check above does not return.  Rather it sets `nodeConnections=0` (and supplements `err`).  So this `if nodeConnections == 0` can happen if (1) `GetConnectionCount` just returns zero node connections, or (2) if the RPC call has an error.  We don't return above in the non-nil error case because we still want to `c.Status.SetConnections(nodeConnections)`.  Since this is somewhat tortuous logic, I'll rewrite this part as follows for clarity:

```go
	nodeConnections, err := c.nodeClient.GetConnectionCount()
	if err != nil {
		// Assume there arr no connections if RPC had an error.
		c.Status.SetConnections(0)
		return fmt.Errorf("failed to get connection count: %v", err)
	}

	// Before updating connections, get the previous connection count.
	prevConnections := c.Status.NodeConnections()

	c.Status.SetConnections(nodeConnections)
	if nodeConnections == 0 {
		return nil
	}
```

Reply:
Oh, right. Thanks.